x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/elastic/ElasticInferenceServiceRequest.java

```diff
@@ -7,6 +7,30 @@
 
 package org.elasticsearch.xpack.inference.external.request.elastic;
 
+import org.apache.http.client.methods.HttpRequestBase;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.xpack.inference.external.request.HttpRequest;
 import org.elasticsearch.xpack.inference.external.request.Request;
 
-public interface ElasticInferenceServiceRequest extends Request {}
+public abstract class ElasticInferenceServiceRequest implements Request {
+
+    private final String productOrigin;
+
+    public ElasticInferenceServiceRequest(String productOrigin) {
+        this.productOrigin = productOrigin;
+    }
+
+    public String getProductOrigin() {
+        return productOrigin;
+    }
+
+    @Override
+    public final HttpRequest createHttpRequest() {
```

Comment:
Made this `final` so it's clear that a new subclass of `ElasticInferenceServiceRequest` should override `createHttpRequestBase` and not `createHttpRequest`.